File: py/2021/day1.py
def read_input(year: int, day: int):
    with open(f"input/{year}/{day}", "r") as f:
        return f.read()


inputs = [int(x) for x in read_input(2021, 1).split("\n")]


def part_one():
    increase_counter = 0
    for i in range(0, len(inputs) - 1):
        if inputs[i] < inputs[i + 1]:
            increase_counter += 1
    print(f"Part One: {increase_counter}")


def part_two():
    increase_counter = 0
    for i in range(0, len(inputs) - 3):
        if (inputs[i] + inputs[i + 1] + inputs[i + 2]) < (
            inputs[i + 1] + inputs[i + 2] + inputs[i + 3]
        ):
            increase_counter += 1
    print(f"Part Two: {increase_counter}")


part_one()
part_two()

File: jl/2021/day1.jl
function parse_input(year, day)
    [parse(Int64, x) for x in split(open(f->read(f, String), "input/$year/$day"), "\n")]
end

inputs = parse_input(2021, 1)

sums = [a + b + c for (a, b, c) ∈ zip(inputs[1:end-2], inputs[2:end-1], inputs[3:end])]
(partone, parttwo) = [count(diff(inputs) .> 0), count(diff(sums) .> 0)]

println("Part One: $partone\nPart Two: $parttwo")
